packages/picasso/src/TopBarMenu/TopBarMenu.tsx

```diff
@@ -0,0 +1,56 @@
+import { BaseProps } from '@toptal/picasso-shared'
+import React, {
+  forwardRef,
+  HTMLAttributes,
+  ReactNode,
+  useEffect,
+  useContext,
+} from 'react'
+import { Theme, makeStyles } from '@material-ui/core/styles'
+
+import Menu from '../Menu'
+import {
+  PageHamburgerContext,
+  PageHamburgerContextProps,
+} from '../PageHamburger'
+import styles from './styles'
+
+export interface Props extends BaseProps, HTMLAttributes<HTMLUListElement> {
+  /** Menu content */
+  children: ReactNode
+}
+
+const useStyles = makeStyles<Theme>(styles, { name: 'PicassoTopBarMenu' })
+
+export const TopBarMenu = forwardRef<HTMLUListElement, Props>(
+  function TopBarMenu(props, ref) {
+    const { children, ...rest } = props
+    const classes = useStyles()
+    const { setShowHamburger } =
+      useContext<PageHamburgerContextProps>(PageHamburgerContext)
+
+    // not more than 6 menu items are allowed
```

Comment:
nit: it looks clear but should we mention the reasoning?

```suggestion
    // not more than 6 menu items are allowed by BASE design
```

Reply:
done